NBitcoin/TransactionBuilder.cs

```diff
@@ -1153,7 +1153,10 @@ public TransactionBuilder Shuffle()
 
 		private void DoShuffle()
 		{
-			if (ShuffleRandom != null)
+			TransactionBuildingContext ctx = new TransactionBuildingContext(this);
+			if (ShuffleRandom != null &&
+				ShuffleInputs && ctx.CanShuffleInputs &&

```

Comment:
remove this, this DoShuffle only shuffle outputs.

Reply:
then let's rename DoShuffle to DoShuffleOutputs?